src/main/java/com/teragrep/pth_06/planner/walker/conditions/ElementCondition.java

```diff
@@ -119,18 +115,39 @@ public Condition condition() {
                 IndexStatementCondition indexStatementCondition = new IndexStatementCondition(
                         value,
                         config,
-                        new Tokenizer(32)
+                        new Tokenizer(0),
+                        condition,
+                        bloomTermId
                 );
                 condition = indexStatementCondition.condition();
+                List<Table<?>> newMatches = indexStatementCondition.matchList();
+                if (!newMatches.isEmpty()) {
+                    tableSet.addAll(newMatches);
+                }
             }
         }
-        if (condition.equals(DSL.noCondition())) {
+        // bloom search can return condition unmodified
+        if (condition.equals(DSL.noCondition()) && !isBloomSearchCondition()) {
             throw new IllegalStateException("Unsupported Element tag " + tag);
         }
         LOGGER.debug("Query condition: <{}>", condition);
         return condition;
     }
 
+    public boolean isBloomSearchCondition() {
+        final String tag = element.tag();
+        final String operation = element.operation();
+        return "indexstatement".equalsIgnoreCase(tag) && "EQUALS".equals(operation) && !config.streamQuery()
+                && config.bloomEnabled();
+    }
+
+    public Set<Table<?>> patternMatchTables() {
```

Comment:
why this is present in both, ElementCondition and the walker:
https://github.com/teragrep/pth_06/pull/87/files#diff-58946539ef28e538e946fbe84b99864bf3f4a912e9dfab8b94f4a206c21bbbf0R103

Reply:
ElementCondition finds pattern match tables from a single element, while walker keeps list of pattern match tables from all traversed elements